### PR TITLE
Blueprint tree always starts at the origin now, "projected" paths are called out explicitly

### LINKED ISSUE #4156: Update DataResultTree to start at origin and introduce "projected" subspace
As part of https://github.com/rerun-io/rerun/issues/4899 we need to update the way that the DataResultTree is presented in the UI:
 - The top-most entity in the group should always be the `$origin` of the SpaceView.
 - Any items that are not part of the $origin should instead be in a separate "Projected" sub-space that starts at "/" instead of $originl.

Mock-up of future UI:
image

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Direct follow up of #5326
* Fixes #4156 

https://github.com/rerun-io/rerun/assets/1220815/77ed6a42-50db-4ca3-b596-8dcf5bce9baf

Follows the design proposal from the ticket directly. Some detail decisions on how things are exactly handled, but code should be a bit more composable now (albeit admittedly still a bit messy)

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/5342/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/5342/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/5342/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/5342)
- [Docs preview](https://rerun.io/preview/6011f651a1ebd432d77d25d3edd7ea746053fd99/docs) 
- [Examples preview](https://rerun.io/preview/6011f651a1ebd432d77d25d3edd7ea746053fd99/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

**Review by abey79 — APPROVED** [review]
I really like the general direction where this is going! I gave it a try and generally worked as expected. I don't have a firm grasp on every single diff in the query code, but, appart from the few suprises I commented, this lgtm :+1:

File: crates/re_viewport/src/space_view_entity_picker.rs [hunk 1]
```diff
@@ -161,7 +161,7 @@ fn add_entities_line_ui(
 
         let add_info = entities_add_info.get(entity_path).unwrap();
 
-        let is_explicitly_excluded = entity_path_filter.is_explicitly_excluded(entity_path);
+        let is_explicitly_excluded = entity_path_filter.is_included(entity_path);
```
[the remark this answers is not in the document]
Comment (Wumpf) [reply crates/re_viewport/src/space_view_entity_picker.rs:164]: something weird is going on here yes. Need to fix

**Comment by Wumpf** [comment]
@abey79 looks like you spotted some severe weirdness in here, thanks! Need to definitely check on these before merging

**Review by abey79 — APPROVED** [review]
Funny these typos didn't actually break more things :)

**Pull request merged**